cmd/kflex/main.go

```diff
@@ -140,8 +146,32 @@ var createCmd = &cobra.Command{
 	},
 }
 
+var adoptCmd = &cobra.Command{
+	Use:   "adopt <name>",
+	Short: "Adopt a control plane from an external cluster",
+	Long: `Adopt a control plane from an external cluster and switches the Kubeconfig context to
+	        the current instance`,
+	Args: cobra.ExactArgs(1),
+	Run: func(cmd *cobra.Command, args []string) {
+		cp := adopt.CPAdopt{
+			CP: common.CP{
+				Ctx:        createContext(),
+				Name:       args[0],
+				Kubeconfig: kubeconfig,
+			},
+			AdoptedKubeconfig:             adoptedKubeconfig,
+			AdoptedContext:                adoptedContext,
+			AdoptedURLOverride:            adoptedURLOverride,
+			AdoptedTokenExpirationSeconds: adoptedTokenExpirationSeconds,
+			SkipURLOverride:               skipURLOverride,
```

Comment:
You wouldn't have to do this copying if the subcommand options were in a struct that contained the `cobra.Command` rather than (roughly) the other way around.